### PR TITLE
Add plugin test framework doc

### PR DESCRIPTION
**Pull request by wu-sheng**
I am providing the English version of the plugin test framework doc, and add more links to the root doc page.

@arugal @aderm Please review too.

### CODE REVIEW

File: docs/en/guides/Plugin-test.md [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+#### segmentItems
+
+By following the flow of HttpClient case, there should be two segments created.
+1. Segment represents the CaseServlet access. Let's name it as `SegmentA`.
+1. Segment represents the ContextPropagateServlet access. Let's name it as `SegmentB`.
+
+```yml
+segments:
+  - applicationCode: httpclient-case
+    segmentSize: ge 2 # healthcheck可能有多个
```
Comment (zhaoyuguang) [inline docs/en/guides/Plugin-test.md]: Do not use Chinese

Comment (wu-sheng) [reply docs/en/guides/Plugin-test.md]: Miss change :)

File: docs/en/guides/Plugin-test.md [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+
+**runningMode** option description.
+
+| Option | description
+| --- | ---
+| default | Active all plugins in `plugin` folder like the official distribution agent. 
+| with_optional | Active `default` and plugins in `optional-plugin` by the give selector.
+| with_bootstrap | Active `default` and plugins in `bootstrap-plugin` by the give selector.
+
+with_optional/with_bootstap supports multiple selectors, separated by `;`.
```
Comment (zhaoyuguang) [inline docs/en/guides/Plugin-test.md]: Is this word `with_bootstrap`?

Comment (wu-sheng) [reply docs/en/guides/Plugin-test.md]: Yes

**Comment by arugal** [comment]
LGTM

File: docs/en/guides/Plugin-test.md [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+
+Every plugin maintained in the main repo is required having its test cases, also matching the versions in the supported list doc. 
+
+## Environment Requirements
+
+1. MacOS/Linux
+2. jdk 8+
+3. Docker
+4. Docker Compose
+
```
Comment (aderm) [inline docs/en/guides/Plugin-test.md]: does have maven?

File: docs/en/guides/Plugin-test.md [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+## Case Base Image Introduction
+
+The test framework provides `JVM-container` and `Tomcat-container` base images. You could choose the suitable one for your test case, if either is suitable, **recommend choose `JVM-container`**.
+
+### JVM-container Image Introduction
+
+[JVM-container](../../../test/plugin/containers/jvm-container) uses `openjdk:8` as the basic image.
+The test case project is required to be packaged as `project-name.zip`, including `startup.sh` and uber jar, by using `mvn package`.
```
Comment (aderm) [inline docs/en/guides/Plugin-test.md]: "mvn package" may change be "mvn clean package"

File: docs/en/guides/Plugin-test.md [hunk 5]
```diff
@@ -0,0 +1,596 @@
...
+
+Take the following test projects as good examples
+* [sofarpc-scenario](../../../test/plugin/scenarios/sofarpc-scenario) as a single project case.
+* [webflux-scenario](../../../test/plugin/scenarios/webflux-scenario) as a case including multiple projects.
+
+### Tomcat-container Image Introduction
+
+[Tomcat-container](../../../test/plugin/containers/tomcat-container) uses `tomcat:8.5.42-jdk8-openjdk` as the basic image.
+The test case project is required to ba packaged as `project-name.war` by using `mvn package`.
```
Comment (aderm) [inline docs/en/guides/Plugin-test.md]: "ba" may be "be"

**Comment by wu-sheng** [comment]
A typo for sure. clean is OK because it is not related to test. Yoi could submit another pr to fix this. The PR has been merged.

File: docs/en/guides/Plugin-test.md [hunk 6]
```diff
@@ -0,0 +1,596 @@
...
+          endTime: nq 0
+          isError: false
+          spanLayer: Http
+          spanType: Exit
+          componentName: null
+          componentId: eq 2
+          tags:
+            - {key: url, value: 'http://127.0.0.1:8080/httpclient-case/case/context-propagate'}
+            - {key: http.method, value: GET}
+          logs: []
```
Comment (aderm) [inline docs/en/guides/Plugin-test.md]: No log is found in the configuration now, the same is true below.

File: docs/en/guides/Plugin-test.md [hunk 7]
```diff
@@ -0,0 +1,596 @@
...
+
+First of all, the test case project could be compiled successfully, with right project structure and be able to deploy.
+The developer should test the start script could run in Linux/MacOS, and entryService/health services are able to provide
+the response.
+
+You could run test by using following commands
+
+```bash
+cd ${SKYWALKING_HOME}
+bash ./test/pugin/run.sh -f ${scenario_name}
```
Comment (aderm) [inline docs/en/guides/Plugin-test.md]: -f parameters feel need to be explained, usually do not need -f

**Pull request merged**